packages/peregrine/lib/talons/CheckoutPage/AddressBook/useAddressBook.js

```diff
@@ -0,0 +1,138 @@
+    } = useQuery(getCustomerCartAddressQuery);
+
+    useEffect(() => {
+        if (customerAddressesError) {
+            console.error(customerAddressesError);
+        }
+
+        if (customerCartAddressError) {
+            console.error(customerCartAddressError);
+        }
+    }, [customerAddressesError, customerCartAddressError]);
+
+    const isLoading =
+        customerAddressesLoading ||
+        customerCartAddressLoading ||
+        setCustomerAddressOnCartLoading;
+
+    const customerAddresses =
+        (customerAddressesData && customerAddressesData.customer.addresses) ||
+        [];
+
+    useEffect(() => {
+        if (customerAddresses.length !== addressCount.current) {
+            // Auto-select newly added address when count changes
+            if (addressCount.current) {
+                const newestAddress =
+                    customerAddresses[customerAddresses.length - 1];
+                setSelectedAddress(newestAddress.id);
+            }
+
+            addressCount.current = customerAddresses.length;
+        }
+    }, [customerAddresses]);
+
+    const handleEditAddress = useCallback(
+        address => {
+            setActiveAddress(address);
+            toggleDrawer('shippingInformation.edit');
+        },
+        [toggleDrawer]
+    );
+
+    const handleAddAddress = useCallback(() => {
+        handleEditAddress();
+    }, [handleEditAddress]);
+
+    const handleSelectAddress = useCallback(addressId => {
+        setSelectedAddress(addressId);
+    }, []);
+
+    if (
+        customerAddresses.length &&
+        customerCartAddressData &&
+        !selectedAddress
+    ) {
+        const { customerCart } = customerCartAddressData;
+        const { shipping_addresses: shippingAddresses } = customerCart;
+        if (shippingAddresses.length) {
+            const primaryCartAddress = shippingAddresses[0];
+
+            const foundSelectedAddress = customerAddresses.find(
+                customerAddress =>
+                    customerAddress.street[0] === primaryCartAddress.street[0]
+            );
+
+            if (foundSelectedAddress) {
+                setSelectedAddress(foundSelectedAddress.id);
+            }
+        }
+    }
+
+    const handleApplyAddress = useCallback(async () => {
+        try {
+            await setCustomerAddressOnCart({
+                variables: {
+                    cartId,
+                    addressId: selectedAddress
+                }
+            });
+        } catch (error) {
+            console.error(error);
+        }
+
+        toggleActiveContent();
+    }, [
+        cartId,
+        selectedAddress,
+        setCustomerAddressOnCart,
+        toggleActiveContent
+    ]);
+
+    return {
+        activeAddress,
+        customerAddresses,
+        isLoading,
+        handleAddAddress,
+        handleApplyAddress,
+        handleSelectAddress,
+        handleEditAddress,
+        selectedAddress,
+        setSelectedAddress
```

Comment:
You can remove this prop as it's not used anywhere else from what I can tell.

Reply:
Removed in b5aec84.